openhcl/openhcl_boot/src/arch/x86_64/tdx.rs

```diff
@@ -94,3 +96,31 @@ impl minimal_rt::arch::IoAccess for TdxIoAccess {
         let _ = tdcall::tdcall_io_out(&mut TdcallInstruction, port, data as u32, 1);
     }
 }
+
+/// Reads MSR using TDCALL
+pub fn read_msr_tdcall(msr_index: u32) -> u64 {
+    let mut msr_value: u64 = 0;
+    tdcall_rdmsr(&mut TdcallInstruction, msr_index, &mut msr_value).unwrap();
+    msr_value
+}
+
+/// Global variable to store tsc frequency.
+static mut TSC_FREQUENCY: u64 = 0;
```

Comment:
Please use `SingleThreaded` like we do in other parts of the bootshim, so you can avoid the unsafe block.